src/MarginTrading.SettingsService/Extensions/TraceabilityApiExtensions.cs

```diff
@@ -0,0 +1,30 @@
+using System;
+using MarginTrading.SettingsService.Contracts.Common;
+
+namespace MarginTrading.SettingsService.Extensions
+{
+    public static class TraceabilityApiExtensions
+    {
+        public static void Validate(this TraceableRequestParams traceability)
+        {
+            if (traceability == null)
+            {
+                throw new ArgumentNullException(nameof(traceability));
+            }   
+        }
+        
+        public static string ExtractCorrelationId(this TraceableRequestParams @params)
+        {
+            return string.IsNullOrWhiteSpace(@params.CorrelationId)
+                ? @params.Id
```

Comment:
I think this is wrong and may lead to confusion. Same for causation ID below. Happy to discuss.